libraries/Filter/LowPassFilter.cpp

```diff
@@ -12,134 +10,134 @@
-
-template <class T>
-void DigitalLPF<T>::compute_alpha(float sample_freq, float cutoff_freq) {
-    if (sample_freq <= 0) {
-        alpha = 1;
-    } else {
-        alpha = calc_lowpass_alpha_dt(1.0/sample_freq, cutoff_freq);
+        output = sample;
     }
+    return output;
 }
 
 // get latest filtered value from filter (equal to the value returned by latest call to apply method)
 template <class T>
 const T &DigitalLPF<T>::get() const {
-    return _output;
+    return output;
 }
 
+// Reset filter to given value
 template <class T>
-void DigitalLPF<T>::reset(T value) { 
-    _output = value;
+void DigitalLPF<T>::reset(const T &value) { 
+    output = value;
     initialised = true;
 }
-    
-////////////////////////////////////////////////////////////////////////////////////////////
-// LowPassFilter
-////////////////////////////////////////////////////////////////////////////////////////////
 
-// constructors
+// Set reset flag such that the filter will be reset to the next value applied
 template <class T>
-LowPassFilter<T>::LowPassFilter() :
-    _cutoff_freq(0.0f) {}
+void DigitalLPF<T>::reset() {
+    initialised = false;
+}
 
-template <class T>
-LowPassFilter<T>::LowPassFilter(float cutoff_freq) :
-    _cutoff_freq(cutoff_freq) {}
+template class DigitalLPF<float>;
+template class DigitalLPF<Vector2f>;
+template class DigitalLPF<Vector3f>;
+
+////////////////////////////////////////////////////////////////////////////////////////////
+// Low pass filter with constant time step
+////////////////////////////////////////////////////////////////////////////////////////////
 
+// constructor
 template <class T>
-LowPassFilter<T>::LowPassFilter(float sample_freq, float cutoff_freq)
+LowPassFilter_const_dt<T>::LowPassFilter_const_dt(const float &sample_freq, const float &new_cutoff_freq)
 {
-    set_cutoff_frequency(sample_freq, cutoff_freq);
+    set_cutoff_frequency(sample_freq, new_cutoff_freq);
 }
 
 // change parameters
 template <class T>
-void LowPassFilter<T>::set_cutoff_frequency(float cutoff_freq) {
-    _cutoff_freq = cutoff_freq;
+void LowPassFilter_const_dt<T>::set_cutoff_frequency(const float &sample_freq, const float &new_cutoff_freq) {
+    cutoff_freq = new_cutoff_freq;
+
+    if (sample_freq <= 0) {
+        alpha = 1;
+    } else {
+        alpha = calc_lowpass_alpha_dt(1.0/sample_freq, cutoff_freq);
+    }
 }
 
+// return the cutoff frequency
 template <class T>
-void LowPassFilter<T>::set_cutoff_frequency(float sample_freq, float cutoff_freq) {
-    _cutoff_freq = cutoff_freq;
-    _filter.compute_alpha(sample_freq, cutoff_freq);
+float LowPassFilter_const_dt<T>::get_cutoff_freq(void) const {
+    return cutoff_freq;
 }
 
-// return the cutoff frequency
+// add a new raw value to the filter, retrieve the filtered result
 template <class T>
-float LowPassFilter<T>::get_cutoff_freq(void) const {
-    return _cutoff_freq;
+T LowPassFilter_const_dt<T>::apply(const T &sample) {
+    return this->_apply(sample, alpha);
```

Comment:
using `this->` is a little odd, its due to class template inheritance. see: https://kernhanda.github.io/templates-inheritance-methods/